### DISCUSSION TITLE
How do I return complex structs properly?

### DISCUSSION THREAD

**Discussion by Tim-Barton** [wailsapp/wails#2034] [Q&A] [answered]
If I'm reading the docs right if I have code like this

```
type GetProjectResp struct {
	test string `json:"test"`
}

func (a *App) GetProjects() GetProjectResp {
	return GetProjectResp{"test"}
}
```

Then when calling `GetProjects` I should get a `GetProjectResp` with a `test` value of `"test"`.
When looking through the models after running a `wails dev` a TS model for GetProjectResp is created, so the generated object stuff seems to be fine.

The issue is that nothing is returned from the backend, and when I inspect the network traffic in Chrome Dev Tools, the ipc websocket states that the returned 'result' is '{}'
Returning strings or bools are both fine

Since this seems like a pretty big deal if it was a bug and no-ones raised anything, I'm assuming I'm just missing something fundamental in terms of what I need to be doing.

Thank you for any insight or suggestions

Wails doctor output in case that's helpful

System
------
OS:		MacOS
Version: 	13.0
ID:		22A380
Go Version:	go1.18.4
Platform:	darwin
Architecture:	amd64

Wails
------
Version: 	v2.1.0

Dependency			Package Name	Status		Version
----------			------------	------		-------
xcode command line tools  	N/A 		Installed 	2396
npm  				N/A 		Installed 	7.24.0
*upx  				N/A 		Available
*nsis  				N/A 		Available

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!
Optional package(s) installation details:
  - upx : Available at https://upx.github.io/
  - nsis : Available at https://nsis.sourceforge.io/Download

**Comment by leaanthony** [comment, marked as answer]
Hi 👋 Go rules around json encoding are that only exported fields are considered, so this should work:
```go
type GetProjectResp struct {
	Test string `json:"test"`
}

func (a *App) GetProjects() GetProjectResp {
	return GetProjectResp{"test"}
}
```

**Reply by Tim-Barton** [reply]
🤦‍♂️ I knew it had to be something simple and stupid I was missing.

Thanks!
